src/Module/Admin/Addons/Details.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Friendica\Module\Admin\Addons;
+
+use Friendica\Content\Text\Markdown;
+use Friendica\Core\Addon;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Database\DBA;
+use Friendica\Module\BaseAdminModule;
+use Friendica\Util\Strings;
+
+class Details extends BaseAdminModule
+{
+	public static function post()
+	{
+		parent::post();
+
+		$a = self::getApp();
+
+		if ($a->argc > 2) {
+			// @TODO: Replace with parameter from router
+			$addon = $a->argv[2];
+			$addon = Strings::sanitizeFilePathItem($addon);
+			if (is_file('addon/' . $addon . '/' . $addon . '.php')) {
+				include_once 'addon/' . $addon . '/' . $addon . '.php';
+				if (function_exists($addon . '_addon_admin_post')) {
+					$func = $addon . '_addon_admin_post';
+					$func($a);
+				}
+
+				$a->internalRedirect('admin/addons/' . $addon);
+			}
+		}
+
+		$a->internalRedirect('admin/addons');
+	}
+
+	public static function content()
+	{
+		parent::content();
+
+		$a = self::getApp();
+
+		$addons_admin = [];
+		$addonsAdminStmt = DBA::select('addon', ['name'], ['plugin_admin' => 1], ['order' => ['name']]);
+		foreach (DBA::toArray($addonsAdminStmt) as $addon) {

```

Comment:
Since the last storage backend move memory issue, we now know that we are using buffered queries whether we're using `DBA::fetch` or `DBA::toArray`, so the memory footprint is the same in either case. If you need a refresher: https://github.com/friendica/friendica/issues/6778#issuecomment-473609964

Reply:
Well, with ```DBA::toArray``` the data is *additionally* loaded into a PHP array.

Reply:
Done.